java/com/fortech/internship/dao/AddressDAOImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.fortech.internship.dao;
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.EntityManager;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+
+import com.fortech.internship.dto.AddressDTO;
+import com.fortech.internship.model.Address;
+import com.fortech.internship.model.User;
+@Component
+@Repository
```

Comment:
If you have @Repository annotation you don't need the @Component one. They are basically the same.

Reply:
solved, please check